File: cp23/hpi-fun2/comppong/executables/solution-konrad.cpp
#include "bits/stdc++.h"
using namespace std;
int main() {
	ios::sync_with_stdio(false);
	cin.tie(nullptr);
	int w, h;
	cin >> w >> h;
	vector a(h, vector<long long>(w));
	for (auto& x : a)
		for (auto& y : x)
			cin >> y;
	long long ans{0};
	for (int i = 0; i < h; i++) {
		vector<long long> b(w);
		for (int j = i; j < h; j++) {
			transform(begin(b), end(b), begin(a[j]), begin(b), plus<>());
			long long last{0};
			for (const auto& x : b)
				ans = max(ans, last = max(last + x, 0ll));
		}
	}
	cout << ans << '\n';
	return 0;
}

File: cp23/hpi-fun2/comppong/executables/solution-konrad.py
w, h = map(int, input().split())
a = [list(map(int, input().split())) for _ in range(h)]
ans = 0
for i in range(h):
    b = w * [0]
    for row in a[i:]:
        for j, x in enumerate(row):
            b[j] += x
        last = 0
        for x in b:
            last = max(last + x, 0)
            ans = max(ans, last)
print(ans)

